mu/interface/dialogs.py

```diff
@@ -447,7 +447,7 @@ def run_pip(self):
         text area.
         """
         package = self.to_add.pop()
-        args = ['-m', 'pip', 'install', package, '--target',
+        args = ['-m', 'pip', 'install', package, '--system', '--target',
```

Comment:
Based on https://github.com/pypa/pip/issues/3826 it sounds like this is a debian specific flag patched into pip, is that right?
If that is the case and we run this in any other platform this is likely going to return an error. In macOS:
```
$ pip install requests --system
Usage:   
  pip install [options] <requirement specifier> [package-index-options] ...
  pip install [options] -r <requirements file> [package-index-options] ...
  pip install [options] [-e] <vcs project url> ...
  pip install [options] [-e] <local project path> ...
  pip install [options] <archive url/path> ...

no such option: --system
```

Reply:
Look at: 
https://github.com/mu-editor/mu/issues/822

maybe...
'''
f = open("/etc/lsb-release",'r')

f.readlines()


['DISTRIB_ID=LinuxMint\n',
'DISTRIB_RELEASE=19\n',
'DISTRIB_CODENAME=tara\n',
'DISTRIB_DESCRIPTION="FREE_DESKTOP_2019.004c"\n']
'''
and check DISTRIB_ID against 'Debian/Ubuntu/LinuxMint/....' ?
And then add --system flag ?
I can make function like debian_detect() which returns True if it will be Debian or other... and then we could add --system to pip. I know, this will be "a hack', but I cannot see any other way to make it works.

Reply:
But do we understand the original problem? Perhaps there is a different workaround or fix.

The issue linked inside #822 is about mixing the `target` and `user' flags in pip, which is not being done in this file.  Is it possible the `user` flag is being added somewhere else?

Reply:
As I investigated the problem, in Debian's based distros the only way to install libs using pip in users directory (via `--target`) is when you use `--system` option. And the only place I found in source is : @@ -447,7 +447,7 @@ def run_pip(self): - so i decided to place my `--system` there.

Reply:
But the `--system` flag has been added by debian to pip as a patch to install the package in the "system python directory", which is the opposite of what Mu should be doing, as it needs to install the package in a special location.

Based on your error output in #822 it looks like the problem here _could be_ that somehow the final command executed includes both the flags `user` and `target`:

```
File "/usr/lib/python3.6/distutils/command/install.py", line 274, in finalize_options
raise DistutilsOptionError("can't combine user with prefix, "
distutils.errors.DistutilsOptionError: can't combine user with prefix, exec_prefix/home, or install_(plat)base
```

In this case  the `user` flag will try to install the package in the user home directory (not good), and the `target` flag will try to install it in the location indicated by Mu (good, but contradictory to the other flag). Because of this, the `system` flag "fixes" the problem by "cancelling out" the `user` flag.

However, if this is the case, where is the `user` flag coming from? I think that's the thing we need to figure out.

Reply:
Hi, I manage to find a pip.conf file inside .pip folder in my home directory. It has a parameter "user=true" in it. When I changed this parameter to false, I could finally use the -t option on pip.

Reply:
Good find @Saziba! 
Were you installing packages with the default Mu installation, or did you edit the source code as well?